javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -0,0 +1,109 @@
+package java.util.regex
+
+import scalanative.native._, stdlib._, stdio._
+import scalanative.runtime.struct
+
+@extern
+object Regex {
+  def regcomp(regex: Ptr[Int], str: CString, num: Int): Int = extern
+  def regexec(regex: Ptr[Int],
+              str: CString,
+              nmatch: Int,
+              matchposition: Ptr[RegMatch.regmatch_t],
+              num2: Int): Int = extern
+}
+
+case class RegexError(error: Int)
+
+trait RegexPattern {
+  def flags(): Int
+  def pattern(): String
+  override def toString(): String
+}
+
+final class Pattern private (_compiled: Ptr[Int],
+                             _pattern: String,
+                             _flags: Int)
+    extends Serializable
+    with RegexPattern {
+  def flags(): Int                = _flags
+  def compiled(): Ptr[Int]        = _compiled
+  def pattern(): String           = _pattern
+  override def toString(): String = _pattern
+
+  def matcher(input: String): RegexMatcher =
+    new RegexMatcher(this, input, 0, input.length)
+  def split(input: CharSequence): Array[String] =
+    split(input, 0)
+
+  def split(input: CharSequence, limit: Int): Array[String] = {
+    val inputStr = input.toString
+
+    if (inputStr == "") {
+      Array("")
+    } else {
+      val lim     = if (limit > 0) limit else Int.MaxValue
+      val matcher = this.matcher(inputStr)
+      val builder = Array.newBuilder[String]
+      var prevEnd = 0
+      var size    = 0
+      while ((size < lim - 1) && matcher.find()) {
+        if (matcher.end == 0) {} else {
+          builder += inputStr.substring(prevEnd, matcher.start)
+          size += 1
+        }
+        prevEnd = matcher.end
+      }
+      builder += inputStr.substring(prevEnd)
+      val result = builder.result()
+
+      if (limit != 0) {
+        result
+      } else {
+        var actualLength = result.length
+        while (actualLength != 0 && result(actualLength - 1) == "") actualLength -= 1
+
+        if (actualLength == result.length) {
+          result
+        } else {
+          val actualResult = new Array[String](actualLength)
+          System.arraycopy(result, 0, actualResult, 0, actualLength)
+          actualResult
+        }
+      }
+    }
+  }
+  protected override def finalize(): Unit = {
+    // Note that a pattern does not exist without _compiled being alloc-ed, so
+    // this free should never apply to non-alloc-ed memory
+    free(_compiled.asInstanceOf[Ptr[Byte]])
+  }
+}
+
+object Pattern {
+  def compile(regex: String, flags: Int): Pattern = {
+    val compiledRegex = malloc(sizeof[Int] * 100).cast[Ptr[Int]] //Dirtiest hack ever
```

Comment:
It's actually fine. What's the size of `regex_t` typically? You can also just do `malloc(400)` instead of using sizeof, it's not really making code any more clear as we're not allocating array of ints.